oauthlib/oauth2/rfc6749/grant_types/implicit.py

```diff
@@ -239,7 +238,12 @@ def create_token_response(self, request, token_handler):
 
         for modifier in self._token_modifiers:
             token = modifier(token, token_handler, request)
-        self.request_validator.save_token(token, request)
+
+        # In OIDC implicit flow it is possible to have a request_type that does
+        # not include the access_token! In this case there is no need to save a token.
+        if "token" in request.response_type.split():
```

Comment:
Kinda expensive to split the string into a list just to check if token is in it.

Reply:
`split()` can not be removed because:

```
ipdb> 'token' in 'id_token'
True
ipdb> 'token' in ['id_token']
False
```
When using implicit flow:

```
http://127.0.0.1:8000/o/authorize?response_type=id_token&client_id=Ktn0Sh4hO2gA8PKC2aqsauY4ZCxNyIdF1wNfFfJ3&redirect_uri=http://localhost/callback&scope=openid&state=af0ifjsldkj&nonce=n-0S6_WzA2Mj
```
without `split` it return access_token instead of only returning `id_token` as requested.